mesonbuild/modules/pkgconfig.py

```diff
@@ -78,33 +80,39 @@ class GenerateKw(TypedDict):
     listify=True,
 )
 
+
+def _as_str(obj: object) -> str:
+    assert isinstance(obj, str)
+    return obj
+
+
 class DependenciesHelper:
     def __init__(self, state: ModuleState, name: str) -> None:
         self.state = state
         self.name = name
-        self.pub_libs = []
-        self.pub_reqs = []
-        self.priv_libs = []
-        self.priv_reqs = []
-        self.cflags = []
-        self.version_reqs = {}
-        self.link_whole_targets = []
-
-    def add_pub_libs(self, libs) -> None:
-        libs, reqs, cflags = self._process_libs(libs, True)
-        self.pub_libs = libs + self.pub_libs # prepend to preserve dependencies
+        self.pub_libs: T.List[LIBS] = []
+        self.pub_reqs: T.List[str] = []
+        self.priv_libs: T.List[LIBS] = []
+        self.priv_reqs: T.List[str] = []
+        self.cflags: T.List[str] = []
+        self.version_reqs: T.DefaultDict[str, T.Set[str]] = defaultdict(set)
+        self.link_whole_targets: T.List[T.Union[build.CustomTarget, build.CustomTargetIndex, build.StaticLibrary]] = []
+
+    def add_pub_libs(self, libs: T.Sequence[ANY_DEP]) -> None:
+        p_libs, reqs, cflags = self._process_libs(libs, True)
+        self.pub_libs = p_libs + self.pub_libs # prepend to preserve dependencies
         self.pub_reqs += reqs
         self.cflags += cflags
 
-    def add_priv_libs(self, libs) -> None:
-        libs, reqs, _ = self._process_libs(libs, False)
-        self.priv_libs = libs + self.priv_libs
+    def add_priv_libs(self, libs: T.Sequence[ANY_DEP]) -> None:
+        p_libs, reqs, _ = self._process_libs(libs, False)
+        self.priv_libs = p_libs + self.priv_libs
         self.priv_reqs += reqs
 
-    def add_pub_reqs(self, reqs) -> None:
+    def add_pub_reqs(self, reqs: T.Sequence[T.Union[str, dependencies.Dependency]]) -> None:
```

Comment:
This does not rely on covariance, and can be T.List; important, because of the T.Sequence[str] trap.

Reply:
Yeah, actually it does. it gets called in `self._add_lib_dependencies` with `List[Dependency]`

Reply:
Nope, that calls add_pub_libs / add_priv_libs.

It passes mypy as List.

Reply:
```
mesonbuild/modules/pkgconfig.py:254:32: error: Argument 1 to "add_priv_libs" of "DependenciesHelper" has incompatible type "List[Dependency]"; expected "List[Union[Dependency, Union[BuildTarget, CustomTarget, CustomTargetIndex], str]]"
mesonbuild/modules/pkgconfig.py:254:32: note: "List" is invariant -- see https://mypy.readthedocs.io/en/stable/common_issues.html#variance
mesonbuild/modules/pkgconfig.py:254:32: note: Consider using "Sequence" instead, which is covariant
```